File: 0118-pascals-triangle/0118-pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    let results = [[1]];
    for(let i = 1; i < numRows; i++) {
        results[i] = [];
        for(let j = 0; j < i + 1; j++) {
            results[i][j] = (results[i - 1][j] || 0) + (results[i - 1][j -1] || 0) 
        }
    }
    return results;
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        results = [[1]]
        for i in range(numRows - 1):
            temp = [0] + results[-1] + [0]
            row = []
            for j in range(len(results[-1]) + 1):
                row.append(temp[j] + temp[j + 1])
            results.append(row)
        return results